backend/substrapp/tasks/k8s_backend.py

```diff
@@ -100,6 +101,13 @@ def watch_pod(name, watch_init_container=False):
                 pretty=True
             )
 
+            # Handle pod error not linked with containers
+            if api_response.status.phase == 'Failed' or (api_response.status.reason and
+               'Evicted' in api_response.status.reason):
+                error_pod = f'Pod failed reason {api_response.status.reason})'
+                logger.error(f'Status for pod "{name}" {api_response.status.phase.lower()} status')
+                finished = True
```

Comment:
Does it make sense to continue the execution inside the while loop if `finished` is set to true? At first sight it seems we would like to directly exit the while loop.

Reply:
Yes, you're right we can early exit with a break or continue :) (as we set finished to true)

Reply:
Thanks!